### ISSUE TITLE
Can't install .[dev] on Arch Linux due to `blspy` and system-wide GMP feud

### ISSUE THREAD

**Issue by veox** [ethereum/trinity#1130]
* Trinity version: `v0.1.0-alpha.28-2-gf5bef2a5` (commit f5bef2a51d224d519bd3d9b8a686c3806a9ec0be)
* OS: Arch Linux
* Python version (output of `python --version`): Python 3.7.4
* Environment (output of `pip freeze`): `I CANNOT`

(Can't provide actual `pip` environment, because it's setting up this environment what fails.)

### What is wrong?

Doing a clean-room install as [described in docs](https://trinity-client.readthedocs.io/en/latest/contributing.html#setting-the-stage) is not possible on Arch: while building the `blspy` wheel, `cmake` tries to use system-wide `libgmp.so`, which is of "unrecognised format". Snippet:

```
$ pip install -e .\[dev\]
...
    Scanning dependencies of target blspy
    [ 97%] Building CXX object src/CMakeFiles/runbench.dir/test-bench.cpp.o
    [ 97%] Building CXX object src/CMakeFiles/runtest.dir/test.cpp.o
    [ 98%] Building CXX object python-bindings/CMakeFiles/blspy.dir/pythonbindings.cpp.o
    /usr/bin/ar: /usr/lib/libgmp.so: file format not recognized
    make[2]: *** [src/CMakeFiles/combined_custom.dir/build.make:60: src/CMakeFiles/combined_custom] Error 1
    make[1]: *** [CMakeFiles/Makefile2:284: src/CMakeFiles/combined_custom.dir/all] Error 2
    make[1]: *** Waiting for unfinished jobs....
    [ 99%] Linking CXX executable runbench
...
```

@jannikluhn has already reported this upstream as issue https://github.com/Chia-Network/bls-signatures/issues/68, which has since been closed without explicit reason (most likely "because distro-specific").

The simplest work-around is to (re)move `/usr/lib/libgmp.so` and `/usr/lib/libsodium.so` (which are symlinks to actual shared library objects) for the duration the wheel is built. The package is built without dependencies on those libraries then. (The built wheel is stored in `pip`'s local cache, so there's no need to repeat the link-shuffling later on.)

### How can it be fixed

Not sure.

Still looks like an upstream issue to me - just not sure _which_ upstream...

**Comment by ralexstokes** [comment]
FYI @ChihChengLiang

**Comment by ChihChengLiang** [comment]
I would say let's remove `blspy` for now, and decide if we want it back when we are switching to the standardized BLS spec.

**Comment by ChihChengLiang** [comment]
closed by #1176
